File: php/src/Models/PostgresConfigIn.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

class PostgresConfigIn implements \JsonSerializable
{
    private array $setFields = [];

    /**
     * @param string $url PostgreSQL connection URL, e.g. `postgres://user@host:5432/dbname?sslmode=require`.
     *
     * Do NOT embed a password here; use the `password` field instead.
     * @param string|null $password  password for the connection
     * @param string      $tableName Table to insert into. May be schema-qualified (e.g. `public.events`).
     *
     * Quote characters are not supported. Each dot-separated segment is automatically double-quoted when the query is built, so `public.events` becomes `"public"."events"`.
     * @param string|null $sslRootCert PEM-encoded CA certificate used to verify the Postgres server's TLS certificate.
     *
     * Supply this to trust a private or self-signed CA when connecting with `sslmode=verify-ca` or `sslmode=verify-full`. Without it, only the built-in public roots are trusted.
     */
    private function __construct(
        public readonly string $url,
        public readonly string $tableName,
        public readonly ?string $password = null,
        public readonly ?string $sslRootCert = null,
        array $setFields = [],
    ) {
        $this->setFields = $setFields;
    }

    /**
     * Create an instance of PostgresConfigIn with required fields.
     */
    public static function create(
        string $url,
        string $tableName,
    ): self {
        return new self(
            url: $url,
            password: null,
            tableName: $tableName,
            sslRootCert: null,
            setFields: ['url' => true, 'tableName' => true]
        );
    }

    public function withPassword(?string $password): self
    {
        $setFields = $this->setFields;
        $setFields['password'] = true;

        return new self(
            url: $this->url,
            password: $password,
            tableName: $this->tableName,
            sslRootCert: $this->sslRootCert,
            setFields: $setFields
        );
    }

    public function withSslRootCert(?string $sslRootCert): self
    {
        $setFields = $this->setFields;
        $setFields['sslRootCert'] = true;

        return new self(
            url: $this->url,
            password: $this->password,
            tableName: $this->tableName,
            sslRootCert: $sslRootCert,
            setFields: $setFields
        );
    }

    public function jsonSerialize(): mixed
    {
        $data = [
            'url' => $this->url,
            'tableName' => $this->tableName];

        if (isset($this->setFields['password'])) {
            $data['password'] = $this->password;
        }
        if (isset($this->setFields['sslRootCert'])) {
            $data['sslRootCert'] = $this->sslRootCert;
        }

        return \Svix\Utils::newStdClassIfArrayIsEmpty($data);
    }

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return new self(
            url: \Svix\Utils::deserializeString($data, 'url', true, 'PostgresConfigIn'),
            password: \Svix\Utils::deserializeString($data, 'password', false, 'PostgresConfigIn'),
            tableName: \Svix\Utils::deserializeString($data, 'tableName', true, 'PostgresConfigIn'),
            sslRootCert: \Svix\Utils::deserializeString($data, 'sslRootCert', false, 'PostgresConfigIn')
        );
    }

    /**
     * Create an instance from a json string.
     */
    public static function fromJson(string $json): self
    {
        $data = json_decode(json: $json, associative: true, depth: 512, flags: JSON_THROW_ON_ERROR);

        return self::fromMixed($data);
    }
}

File: java/lib/src/main/java/com/svix/models/PostgresConfigIn.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.svix.Utils;

import lombok.EqualsAndHashCode;
import lombok.ToString;

@ToString
@EqualsAndHashCode
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(getterVisibility = Visibility.NONE, setterVisibility = Visibility.NONE)
public class PostgresConfigIn {
    @JsonProperty private String url;
    @JsonProperty private String password;
    @JsonProperty private String tableName;
    @JsonProperty private String sslRootCert;

    public PostgresConfigIn() {}

    public PostgresConfigIn url(String url) {
        this.url = url;
        return this;
    }

    /**
     * PostgreSQL connection URL, e.g. `postgres://user@host:5432/dbname?sslmode=require`.
     *
     * <p>Do NOT embed a password here; use the `password` field instead.
     *
     * @return url
     */
    @javax.annotation.Nonnull
    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public PostgresConfigIn password(String password) {
        this.password = password;
        return this;
    }

    /**
     * Password for the connection.
     *
     * @return password
     */
    @javax.annotation.Nullable
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public PostgresConfigIn tableName(String tableName) {
        this.tableName = tableName;
        return this;
    }

    /**
     * Table to insert into. May be schema-qualified (e.g. `public.events`).
     *
     * <p>Quote characters are not supported. Each dot-separated segment is automatically
     * double-quoted when the query is built, so `public.events` becomes `"public"."events"`.
     *
     * @return tableName
     */
    @javax.annotation.Nonnull
    public String getTableName() {
        return tableName;
    }

    public void setTableName(String tableName) {
        this.tableName = tableName;
    }

    public PostgresConfigIn sslRootCert(String sslRootCert) {
        this.sslRootCert = sslRootCert;
        return this;
    }

    /**
     * PEM-encoded CA certificate used to verify the Postgres server's TLS certificate.
     *
     * <p>Supply this to trust a private or self-signed CA when connecting with `sslmode=verify-ca`
     * or `sslmode=verify-full`. Without it, only the built-in public roots are trusted.
     *
     * @return sslRootCert
     */
    @javax.annotation.Nullable
    public String getSslRootCert() {
        return sslRootCert;
    }

    public void setSslRootCert(String sslRootCert) {
        this.sslRootCert = sslRootCert;
    }

    /**
     * Create an instance of PostgresConfigIn given an JSON string
     *
     * @param jsonString JSON string
     * @return An instance of PostgresConfigIn
     * @throws JsonProcessingException if the JSON string is invalid with respect to
     *     PostgresConfigIn
     */
    public static PostgresConfigIn fromJson(String jsonString) throws JsonProcessingException {
        return Utils.getObjectMapper().readValue(jsonString, PostgresConfigIn.class);
    }

    /**
     * Convert an instance of PostgresConfigIn to an JSON string
     *
     * @return JSON string
     */
    public String toJson() throws JsonProcessingException {
        return Utils.getObjectMapper().writeValueAsString(this);
    }
}
